### DISCUSSION TITLE
Fetching app sources Error

### DISCUSSION THREAD

**Discussion by stevenQiang** [codemagic-ci-cd/codemagic-docs#2157] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. Fetching app sources

### Expected results

success git clone 

### Actual results

Previously, I was able to successfully clone the code, but since yesterday, it has been giving me errors, which seem to be related to a network issue. Additionally, when we try to clone it using Git ourselves, it works fine.

error:
```
...
remote: Compressing objects: 100% (3153/3153)        
remote: Compressing objects: 100% (3153/3153), done.        
Receiving objects:   0% (1/5196)
Receiving objects:   0% (40/5196), 12.00 KiB | 1024 bytes/s
Receiving objects:   1% (52/5196), 12.00 KiB | 1024 bytes/s
Receiving objects:   2% (104/5196), 12.00 KiB | 1024 bytes/s
Receiving objects:   2% (122/5196), 28.00 KiB | 3.00 KiB/s  
Receiving objects:   3% (156/5196), 28.00 KiB | 3.00 KiB/s
Receiving objects:   3% (201/5196), 44.00 KiB | 4.00 KiB/s
Receiving objects:   4% (208/5196), 44.00 KiB | 4.00 KiB/s
Receiving objects:   4% (219/5196), 52.00 KiB | 4.00 KiB/s
Receiving objects:   4% (226/5196), 60.00 KiB | 4.00 KiB/s
Receiving objects:   4% (226/5196), 68.00 KiB | 3.00 KiB/s
Receiving objects:   4% (226/5196), 76.00 KiB | 3.00 KiB/s
Receiving objects:   4% (226/5196), 92.00 KiB | 3.00 KiB/s
Receiving objects:   4% (226/5196), 100.00 KiB | 3.00 KiB/s
Timeout, server gitlab.xxxxxx.com not responding.
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output


Build failed :|
Failed to clone repository
```

### Build id (optional)

653b714c67bf55a93868dd27

**Comment by dtrdic** [comment]
hello @stevenQiang, maybe this repository has some additional restrictions and you need to grant Codemagic access to it.. try to create a new ssh key pair for your app, checkout [this guide](https://docs.codemagic.io/knowledge-git/access-private-git-submodules/) on how to do it

**Comment by stevenQiang** [comment, marked as answer]
Thank you for your response. Because I'm in China, I suspected it might be a network issue, so today I added an extra layer of CDN, and now I can clone the code normally. 
I used to be able to do it all the time, so did your builder servers change the network architecture?

**Reply by dtrdic** [reply]
Glad to hear that it's working for you now!:) There haven't been any changes from our side regarding network architecture. I'll close this thread since the issue has been resolved.. have a nice day!
